### PR TITLE
Feature/wb 12/world naming input form panel

### PR DESCRIPTION
**Pull request by Daboo1998**

### CODE REVIEW

File: src/api/constants/app.variables.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ const BASE_URL =
 export const appVariables: AppConfigExtendedInterface = {
   APP_VERSION: 'v0.14.2',
   BACKEND_ENDPOINT_URL: '',
+  BACKEND_V4_ENDPOINT_URL: 'https://dev.odyssey.ninja/api/v4',
```
Comment (PAKovalev) [inline src/api/constants/app.variables.ts:16]: it must come from backend.

File: src/scenes/App.tsx [hunk 2]
```diff
@@ -101,7 +101,13 @@ const App: FC = () => {
   if (!sessionStore.oidcConfig) {
     return (
       <Switch>
-        <Redirect to={{pathname: ROUTES.login, state: {from: pathname}}} />
+        <Redirect
```
Comment (PAKovalev) [inline src/scenes/App.tsx:104]: if (!sessionStore.oidcConfig && pathname === ROUTES.worldBuilder.base) {
  ....
} 

if (!sessionStore.oidcConfig) {
  ....
}

File: src/scenes/App.tsx [hunk 3]
```diff
@@ -123,12 +129,43 @@ const App: FC = () => {
...
+    return (
+      <ThemeProvider theme={themeStore.theme}>
+        <Web3ReactProvider getLibrary={sessionStore.getLibrary}>
+          <AuthProvider {...sessionStore.oidcConfig}>
+            <AppAuth>
+              <GlobalStyles />
+              <AppLayers
+                withUnity={false}
+                withMeeting={false}
+                isWorldBuilder={pathname.includes(ROUTES.worldBuilder.base)}
```
Comment (PAKovalev) [inline src/scenes/App.tsx:154]: Don't use isWorldBuilder name

File: src/scenes/widgets/Widgets.tsx [hunk 4]
```diff
@@ -25,7 +25,11 @@ import {
 import * as styled from './Widgets.styled';
 import {AvatarForm} from './pages/ProfileWidget/components';
 
-const Widgets: FC = () => {
+interface PropsInterface {
+  isWorldBuilder?: boolean;
```
Comment (PAKovalev) [inline src/scenes/widgets/Widgets.tsx:29]: ??

File: src/scenes/widgets/Widgets.tsx [hunk 5]
```diff
@@ -123,41 +129,45 @@ const Widgets: FC = () => {
       />
       <styled.Footer data-testid="Widgets-test">
         <styled.MainLinks>
-          <ToolbarIcon
```
Comment (PAKovalev) [inline src/scenes/widgets/Widgets.tsx:126]: I don't like this `mix`

File: src/scenes/widgets/pages/ProfileMenuWidget/ProfileMenuWidget.tsx [hunk 6]
```diff
@@ -5,7 +5,11 @@ import {useStore} from 'shared/hooks';
 
 import {Menu, Setting, TokenRuleReview, TokenRules} from './components';
 
-const ProfileMenuWidget: FC = () => {
+interface PropsInterface {
+  isWorldBuilder?: boolean;
```
Comment (PAKovalev) [inline src/scenes/widgets/pages/ProfileMenuWidget/ProfileMenuWidget.tsx:9]: It is bad.

File: src/scenes/widgets/pages/ProfileMenuWidget/components/Menu/Menu.tsx [hunk 7]
```diff
@@ -26,7 +30,7 @@ const ProfileMenuWidget: FC = () => {
 
   const signOutUser = async () => {
     await sessionStore.logout(auth);
-    document.location.href = ROUTES.base;
+    document.location.href = isWorldBuilder ? ROUTES.worldBuilder.base : ROUTES.base;
```
Comment (PAKovalev) [inline src/scenes/widgets/pages/ProfileMenuWidget/components/Menu/Menu.tsx:33]: ?

File: src/scenes/widgets/pages/ProfileMenuWidget/components/Menu/Menu.tsx [hunk 8]
```diff
@@ -65,41 +69,47 @@ const ProfileMenuWidget: FC = () => {
...
-          <Text text={profile.status ? t(`labels.${profile.status}`) : ''} size="xxs" />
-        </styled.Option>
-        {profile.isNodeAdmin && (
-          <styled.Option onClick={handleTokenRulesOpen}>
-            <styled.IconContainer>
-              <IconSvg name="whitelist" size="medium-large" isWhite />
-            </styled.IconContainer>
-            <Text text={t('labels.tokenRules')} size="xxs" />
-          </styled.Option>
+        {!isWorldBuilder && (
```
Comment (PAKovalev) [inline src/scenes/widgets/pages/ProfileMenuWidget/components/Menu/Menu.tsx:78]: No please.

**Review by PAKovalev — CHANGES_REQUESTED** [review]

File: src/api/constants/app.variables.ts (hunk 1 above)
Comment (Daboo1998) [reply src/api/constants/app.variables.ts:16]: It's temporary, we remove it when all other BE EP will be migrated to V4 and we will use one BACKEND

**Review by PAKovalev — APPROVED** [review]

**Review by dmitry-yudakov — APPROVED** [review]

**Pull request merged**